### PR TITLE
Playback 2023 - Improvements

### PR DESCRIPTION
**Pull request by ashiagr**
| 📘 Part of: https://github.com/Automattic/pocket-casts-android/issues/1463 | 
|:---:|

## Description

- Update the share copies, hashtag
- Displays app logo in shared image
- Fix text cutoff in year over year story (https://github.com/Automattic/pocket-casts-android/pull/1502#pullrequestreview-1706653781)
- Update epilogue button style (https://github.com/Automattic/pocket-casts-android/pull/1501#pullrequestreview-1706470948)
- Update percentage in the completion story (https://github.com/Automattic/pocket-casts-android/pull/1506#pullrequestreview-1708258927)
- Updates top podcast query in https://github.com/Automattic/pocket-casts-android/pull/1524/commits/7b76ad7b93b507470374520a486b761b78a3372a

#### Update the share copies, hashtag, add app logo

- Run the app on a real device
- Make sure you're logged in to an account that has a few episodes listened to this year and last year
- Go to Profile 
- Open stories
- Test sharing each story
- ✅ #playback2023 hashtag should be shared
- ✅ Double-check the text content
- ✅ Check that shared images have app logo
- ✅ Double-check the text content

Hash tag| Shared image with logo
---|---
| 

#### Fix text cutoff in year over year story, Update epilogue button style, Update percentage in the completion story 

- Edit EndOfYearManagerImpl to return 
  ` YearOverYearListeningTime(totalPlayedTimeThisYear = 10000,  totalPlayedTimeLastYear = 0,)` for `StoryYearOverYear` at line 166
   `EpisodesStartedAndCompleted(started = 0, completed = 0)` for `StoryCompletionRate` at line 167
- Make sure you're logged in to Plus account that has a few episodes listened to this year and last year
- Go to Profile 
- Open stories
- Go to top category story 
- ✅ Check that the top category story has appropriate newlines for english lang.
- Go to year over year story
- ✅ Check that text is not cutoff in year over year story when there is a huge decrease/ increase in the listening time 
- Go to completion rate story
- ✅ Check percentage is zero when no episode was started 
- Go to epilogue story
- ✅ Check that "Play again" button looks good
- ✅ Check that when you tap between stories, title remain stable (except on Plus story it it moves below the subscription badge)

 Top Category| Year over Year| Completion Rate | Epilogue |
----|-----|----|---
| | | 

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">No reviewers have been set for this PR yet. Please request a review from <strong>@ Automattic/pocket-casts-android</strong>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/PodcastDao.kt [hunk 1]
```diff
@@ -354,13 +354,14 @@ abstract class PodcastDao {
...
     @Query(
         """
-        SELECT DISTINCT podcast_episodes.uuid as episodeId, podcasts.uuid, podcasts.title, podcasts.author, podcasts.primary_color as tintColorForLightBg, podcasts.secondary_color as tintColorForDarkBg, SUM(podcast_episodes.played_up_to) as totalPlayedTime, COUNT(podcast_episodes.uuid) as numberOfPlayedEpisodes
-        FROM podcast_episodes
-        JOIN podcasts ON podcast_episodes.podcast_id = podcasts.uuid
-        WHERE podcast_episodes.last_playback_interaction_date IS NOT NULL AND podcast_episodes.last_playback_interaction_date > :fromEpochMs AND podcast_episodes.last_playback_interaction_date < :toEpochMs
-        GROUP BY podcast_id
-        ORDER BY totalPlayedTime DESC, numberOfPlayedEpisodes DESC
-        LIMIT :limit
+         SELECT (totalPlayedTime * 0.4) * (numberOfPlayedEpisodes * 0.6) as weighted, * FROM (
```
Comment (ashiagr) [inline modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/PodcastDao.kt:357]: This matches iOS query: https://github.com/Automattic/pocket-casts-ios/pull/1192/commits/844bd08715e7a6d51466448e6283947c94e3872e for top podcast

Slack discussion: p1698773178757819-slack-C041BGTFQ5R

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by mchowning — APPROVED** [review]
Nice work, these are some good improvements!

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**